### DISCUSSION TITLE
Can I create an onRequest interceptor ?

### DISCUSSION THREAD

**Discussion by JohannDesobry** [flutterdata/flutter_data#243] [General]
I want to verify if the user is logged in before making the request. If not, I will refresh the token and then continue with the request.

**Comment by frank06** [comment]
There is no such thing, we should add it

**Reply by JohannDesobry** [reply]
Would be very much appreciated

**Comment by frank06** [comment]
Ok now that I think about it, have you tried `shouldLoadRemoteAll` / `shouldLoadRemoteOne`? These allow you to cancel a `findAll`/`findOne` request. Override it in your model adapter.

There's also the possibility of overriding `defaultHeaders` which is an async getter. That gives you the chance to refresh a token and let the request proceed with new headers. It would also work with a plain `sendRequest`

**Reply by JohannDesobry** [reply]
I will try with the defaultHeaders, sounds good to me. Thanks
